File: frontend/src/api/tmdb.ts
import api from './axios';
import { ContentType } from '../types';

export interface TmdbSearchResult {
  tmdbId: number;
  titulo: string;
  ano: number | null;
  imagemUrl: string | null;
  sinopse: string | null;
  tipo: ContentType;
  avaliacao: number | null;
}

export interface TmdbImportResult {
  conteudoId: number;
  titulo: string;
  temporadasImportadas: number;
  episodiosImportados: number;
}

export async function getTmdbStatus(): Promise<{ configurado: boolean }> {
  const { data } = await api.get('/tmdb/status');
  return data;
}

export async function searchTmdb(query: string, tipo: ContentType): Promise<TmdbSearchResult[]> {
  const { data } = await api.get<TmdbSearchResult[]>('/tmdb/search', { params: { query, tipo } });
  return data;
}

export async function importTmdb(tmdbId: number, tipo: ContentType): Promise<TmdbImportResult> {
  const { data } = await api.post<TmdbImportResult>('/tmdb/import', null, { params: { tmdbId, tipo } });
  return data;
}

File: backend/app/schemas/tmdb.py
from app.models.enums import ContentType
from app.schemas.base import CamelModel


class TmdbStatus(CamelModel):
    configurado: bool


class TmdbSearchResult(CamelModel):
    tmdb_id: int
    titulo: str
    ano: int | None
    imagem_url: str | None
    sinopse: str | None
    tipo: ContentType
    avaliacao: float | None


class TmdbImportResult(CamelModel):
    conteudo_id: int
    titulo: str
    temporadas_importadas: int
    episodios_importados: int
